Method argument annotations from processor not working

### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. See https://github.com/tomas-langer/reproduce-micronaut-visitor

### Expected Behaviour

When I have a `TypeElementVisitor` implemented that adds an annotation to a method parameter, that annotation should be visible at runtime in an executable method

The reproducer main method should successfully execute.

### Actual Behaviour

The annotation is not visible (nor is it in the generated class of the executable method).

Exception from reproducer:
```
Caused by: io.micronaut.core.bind.exceptions.UnsatisfiedArgumentException: Required argument [FirstBean paramInjection] not specified
	at io.micronaut.core.bind.DefaultExecutableBinder.bind(DefaultExecutableBinder.java:96)
	at io.reproducers.reproducer.impl.ExecMethodProcessor.process(ExecMethodProcessor.java:37)
	at io.reproducers.reproducer.impl.ExecMethodProcessor.process(ExecMethodProcessor.java:17)
	at io.micronaut.context.AnnotationProcessorListener.onCreated(AnnotationProcessorListener.java:89)
	... 10 more
```

### Environment Information

- **Operating System**: MacOS 
- **Micronaut Version:** 2.0.1
- **JDK Version:** 11

### Example Application

https://github.com/tomas-langer/reproduce-micronaut-visitor

--- a/inject-java-test/src/test/groovy/io/micronaut/inject/visitor/ElementAnnotateSpec.groovy
+++ b/inject-java-test/src/test/groovy/io/micronaut/inject/visitor/ElementAnnotateSpec.groovy
@@ -6,10 +6,10 @@ import io.micronaut.annotation.processing.test.JavaParser
 import io.micronaut.aop.Introduction
 import io.micronaut.core.annotation.AnnotationValueBuilder
 import io.micronaut.core.annotation.Introspected
-import io.micronaut.core.beans.BeanIntrospection
 import io.micronaut.inject.BeanDefinition
 import io.micronaut.inject.ast.ClassElement
 import io.micronaut.inject.ast.MethodElement
+import io.micronaut.inject.ast.ParameterElement
 import io.micronaut.inject.ast.PropertyElement
 import io.micronaut.inject.beans.visitor.IntrospectedTypeElementVisitor
 import io.micronaut.inject.writer.BeanDefinitionVisitor
@@ -72,13 +72,17 @@ class TestListener {
 }
 
 ''')
+        def receiveMethod = definition.findMethod("receive", String).get()
+        def vArgument = receiveMethod.getArguments()[0]
 
         expect:
         definition.hasAnnotation(Ann)
         !definition.booleanValue(Ann, "aggregating").orElse(false)
         definition.booleanValue(Ann, "isolating").orElse(false)
         definition.getValue(Ann, "foo", String).get() == 'bar'
-        definition.findMethod("receive", String).get().hasAnnotation(Ann)
+        receiveMethod.hasAnnotation(Ann)
+        vArgument.getAnnotationMetadata().hasAnnotation(Ann)
+        vArgument.getAnnotationMetadata().getValue(Ann, "foo, String").get() == 'bar'
     }
 
     void "test annotation bean introspection properties"() {
@@ -160,6 +164,13 @@ class Test {
         void visitMethod(MethodElement element, VisitorContext context) {
             element.annotate(Ann) { AnnotationValueBuilder builder ->
                 builder.member("foo", "bar")
+            }
+
+            for (ParameterElement parameterElement : element.getParameters()) {
+                parameterElement.annotate(Ann) {
+                    AnnotationValueBuilder builder ->
+                        builder.member("foo", "bar")
+                }
             }
         }
     }

--- a/inject-java-test/src/test/groovy/io/micronaut/inject/visitor/ElementAnnotateSpec.groovy
+++ b/inject-java-test/src/test/groovy/io/micronaut/inject/visitor/ElementAnnotateSpec.groovy
@@ -82,7 +82,7 @@ class TestListener {
         definition.getValue(Ann, "foo", String).get() == 'bar'
         receiveMethod.hasAnnotation(Ann)
         vArgument.getAnnotationMetadata().hasAnnotation(Ann)
-        vArgument.getAnnotationMetadata().getValue(Ann, "foo, String").get() == 'bar'
+        vArgument.getAnnotationMetadata().getValue(Ann, "foo", String).get() == 'bar'
     }
 
     void "test annotation bean introspection properties"() {
@@ -172,6 +172,7 @@ class Test {
                         builder.member("foo", "bar")
                 }
             }
+
         }
     }
 

--- a/inject-java/src/main/java/io/micronaut/annotation/processing/visitor/AbstractJavaElement.java
+++ b/inject-java/src/main/java/io/micronaut/annotation/processing/visitor/AbstractJavaElement.java
@@ -25,6 +25,7 @@ import io.micronaut.inject.ast.ClassElement;
 import io.micronaut.inject.ast.MemberElement;
 
 import edu.umd.cs.findbugs.annotations.NonNull;
+import io.micronaut.inject.ast.ParameterElement;
 import io.micronaut.inject.ast.PrimitiveElement;
 
 import javax.lang.model.element.Element;
@@ -76,7 +77,20 @@ public abstract class AbstractJavaElement implements io.micronaut.inject.ast.Ele
                 .newAnnotationBuilder()
                 .annotate(annotationMetadata, av);
 
-        String declaringTypeName = this instanceof MemberElement ? ((MemberElement) this).getOwningType().getName() : getName();
+        String declaringTypeName;
+        if (this instanceof MemberElement) {
+            declaringTypeName = ((MemberElement) this).getOwningType().getName();
+        } else if (this instanceof ParameterElement) {
+            TypeElement typeElement = visitorContext.getModelUtils().classElementFor((Element) this.getNativeType());
+            if (typeElement == null) {
+                declaringTypeName = getName();
+            } else {
+                declaringTypeName = typeElement.getQualifiedName().toString();
+            }
+        } else {
+            declaringTypeName = getName();
+        }
+
         AbstractAnnotationMetadataBuilder.addMutatedMetadata(declaringTypeName, element, annotationMetadata);
         visitorContext.getAnnotationUtils().invalidateMetadata(element);
         return this;

